SimExLite/PMICalculators/SimpleScatteringPMICalculator.py

```diff
@@ -0,0 +1,11147 @@
+        pmi_scattering.g_s2e["setup"]["num_digits"] = 7
+        pmi_scattering.g_s2e["steps"] = self.parameters["number_of_steps"].value
+        pmi_scattering.g_s2e["maxZ"] = 100
+        pmi_scattering.g_s2e["random_rotation"] = False
+        pmi_scattering.g_s2e["setup"]["pmi_out"] = output_fn
+
+        pmi_scattering.f_dbase_setup()
+        pmi_scattering.f_save_info()
+        pmi_scattering.f_load_pulse(pmi_scattering.g_s2e["prop_out"])
+
+        pmi_scattering.g_s2e["sample"] = SampleData_to_atoms_dict(sample_data)
+        pmi_scattering.f_rotate_sample()
+        pmi_scattering.f_system_setup()
+        pmi_scattering.f_time_evolution()
+
+        assert len(self.output_keys) == 1
+        key = self.output_keys[0]
+        output_data = self.output[key]
+        output_data.set_file(output_fn, XMDYNFormat)
+
+        return self.output
+
+
+def prepare_wavefront_file(wavefront_data: WavefrontData):
+    """Prepare the wavefront file for the backengine"""
+    if wavefront_data.mapping_type == WPGFormat:
+        wavefront_fn = wavefront_data.filename
+    else:
+        temp_name = next(tempfile._get_candidate_names())
+        wavefront_fn = "input_wavefront_" + temp_name + ".h5"
+        wavefront_data.write(wavefront_fn, WPGFormat)
+    return wavefront_fn
+
+
+def SampleData_to_atoms_dict(sample_data: SampleData) -> dict:
+    """Read the sampleData object into an atoms_dict for the backengine"""
+    atoms_dict = {
+        "Z": [],  # Atomic number.
+        "r": [],  # Cartesian coordinates.
+        "selZ": {},  # Abundance of each element.
+        "N": 0,  # Number of atoms.
+    }
+    input_dict = sample_data.get_data()
+    atoms_dict["Z"] = input_dict["atomic_numbers"]
+    atoms_dict["r"] = input_dict["positions"] * 1e-10
+    atoms_dict["N"] = len(input_dict["atomic_numbers"])
+    for sel_Z in numpy.unique(atoms_dict["Z"]):
+        atoms_dict["selZ"][sel_Z] = numpy.nonzero(atoms_dict["Z"] == sel_Z)[0]
+    return atoms_dict
+
+
+class PMIScattering(object):
+    def __init__(self):
+        self.g_s2e = {}
+        self.g_dbase = {}
+
+        self.f_s2e_setup()
+
+    def f_s2e_setup(self):
+        self.g_s2e["setup"] = dict()
+        self.g_s2e["sys"] = dict()
+        self.g_s2e["setup"]["num_digits"] = 7
+        self.g_s2e["steps"] = 100
+        self.g_s2e["maxZ"] = 100
+
+    ##############################################################################
+
+    def f_save_info(self):
+        pmi_file = self.g_s2e["setup"]["pmi_out"]
+
+        grp = "/info"
+        ###    print pmi_file , grp
+        xfp = h5py.File(pmi_file, "a")
+        try:
+            grp_hist_parent = xfp.create_group(grp)
+        except:
+            1
+
+        #    xfp[ grp + '/Sq_free' ] = g_dbase['Sq_free']
+        xfp.close()
+
+    ##############################################################################
+
+    def f_init_random(self):
+        random.seed(self.g_s2e["id"])
```

Comment:
`random` is not imported.

Reply:
`f_init_random` was not called in the calculator. I will just comment the function out to avoid confusion.